src/http/headers.cr

```diff
@@ -222,4 +230,16 @@ struct HTTP::Headers
       values.join ","
     end
   end
+
+  private def check_invalid_header_content(value)
+    # According to RFC 7230, characters accepted as HTTP header
+    # are '\t', ' ', all US-ASCII printable characters and
+    # range from '\x80' to '\xff' (but the last is obsoleted.)
+    value.each_char do |char|
```

Comment:
But the [RFC](https://tools.ietf.org/html/rfc7230#section-3.2.6) talks about octets (bytes). Why we need to decode the string into UTF-8 and then check for char validity?

I understand that if we don't do this we can end up with non UTF-8 strings in memory, but I prefer this to making things a bit slower. What do you think?

(That is, unless I'm misunderstanding something in the RFC)

Another thing is, `single_byte_optimizable` is only known when you invoke `String#size`, until then it will just decode char by char.